### PR TITLE
Add gain and builders to AudioSource

### PR DESCRIPTION
**Pull request by sornas**
This is the start of #33 but we need that #turnaround time.

### CODE REVIEW

File: src/audio.rs [hunk 1]
```diff
@@ -5,15 +5,30 @@ use luminance_sdl2::sdl2::audio::{AudioCallback, AudioDevice, AudioSpecDesired};
...
+                self.$field = $field;
+                self
+            }
+        )*
+    }
+}
+
+/// A sound that is playing or can be played.
+pub struct AudioSource {
     /// Which specific sample we're currently on.
```
Comment (FredTheDino) [inline src/audio.rs:21]: Indent maybe?

File: src/audio.rs [hunk 2]
```diff
@@ -80,7 +117,7 @@ impl AudioCallback for Audio {
                 }
 
                 // Write data
-                *x += samples[source.position];
+                *x += samples[source.position] * source.gain;
```
Comment (FredTheDino) [inline src/audio.rs:120]: Nice!

File: src/audio.rs [hunk 3]
```diff
@@ -44,15 +76,20 @@ impl Audio {
...
-    pub fn play(&mut self, audio: &asset::Audio) {
-        self.sources.push(AudioSource {
-            position: 0,
-            looping: false,
-            samples: audio.samples(),
+    /// Start playing a new sound once.
+    ///
+    /// This uses the "default" settings for the [AudioSource]. If you want to modify it, use
+    /// [Audio::play_source].
+    pub fn play_asset(&mut self, audio: &asset::Audio) {
```
Comment (FredTheDino) [inline src/audio.rs:83]: Not sure i like this.

When would you use it? Why not use the default audio play?

File: src/audio.rs [hunk 4]
```diff
@@ -44,15 +76,20 @@ impl Audio {
...
+        self.play_source(AudioSource::new(audio));
+    }
 
-            remove: false,
-        });
+    /// Start playing a new source.
+    ///
+    /// The source can be created via [AudioSource::new] and modified by builders on [AudioSource]
+    /// (like [AudioSource::looping]).
+    pub fn play_source(&mut self, source: AudioSource) {
```
Comment (FredTheDino) [inline src/audio.rs:91]: Name this "play", since it's the thing we want people to use?

**Review by FredTheDino — APPROVED** [review]
Nice! #TurnAroundTime

File: src/audio.rs (hunk 1 above)
Comment (sornas) [reply src/audio.rs:21]: Indent how? Line 21 is the documentation for line 22.

**Comment by FredTheDino** [comment]
+1

**Pull request merged**